hudi-platform-service/hudi-metaserver/hudi-metaserver-client/src/main/java/org/apache/hudi/common/table/HoodieTableMetaserverClient.java

```diff
@@ -60,18 +60,17 @@ public class HoodieTableMetaserverClient extends HoodieTableMetaClient {
 
   public HoodieTableMetaserverClient(Configuration conf, String basePath, ConsistencyGuardConfig consistencyGuardConfig,
                                      String mergerStrategy, FileSystemRetryConfig fileSystemRetryConfig,
-                                     String databaseName, String tableName, HoodieMetaserverConfig config) {
+                                     HoodieMetaserverConfig config) {
     super(conf, basePath, false, consistencyGuardConfig, Option.of(TimelineLayoutVersion.CURR_LAYOUT_VERSION),
         config.getString(HoodieTableConfig.PAYLOAD_CLASS_NAME), mergerStrategy, fileSystemRetryConfig);
-    checkArgument(nonEmpty(databaseName), "database name is required.");
-    checkArgument(nonEmpty(tableName), "table name is required.");
-    this.databaseName = databaseName;
-    this.tableName = tableName;
+    checkArgument(nonEmpty(tableConfig.getDatabaseName()), "database name is required.");
+    checkArgument(nonEmpty(tableConfig.getTableName()), "table name is required.");
+    this.databaseName = tableConfig.getDatabaseName();
```

Comment:
Where was the table config initialized ?

Reply:
The table config belongs to `HoodieTableMetaClient`, will be initialized first in the `HoodieTableMetaClient`'s constructor. it should not be initialized again.